sdk/src/main/scala/io/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,270 @@
+    rpcParams.forEach(subscriptionId => {
+      subscriptionIdsToRemove.add(Numeric.decodeQuantity(subscriptionId.asText()))
+    })
+    val removedSubscription = WebSocketAccountServerEndpoint.removeSubscriptions(subscriptionIdsToRemove.toArray(new Array[BigInteger](0)))
+    if (!removedSubscription) {
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+        new RpcError(RpcCode.InvalidParams, s"Subscription ID not found.", "")),
+        session)
+    } else
+      WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, true), session)
+  }
+
+  private def validateRpcRequestParams(rpcParams: JsonNode, rpcRequest: RpcId, session: Session): Unit = {
+    if (rpcParams == null || !rpcParams.isArray || rpcParams.size() < 1) {
+      log.debug("Missing or empty field params.")
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest,
+        new RpcError(RpcCode.InvalidParams, "Missing or empty field params.", "")),
+        session)
+    }
+  }
+
+  private def createSubscriptionId(): BigInteger = {
+    BigInteger.valueOf(WebSocketAccountServerEndpoint.subscriptionCounter.incrementAndGet())
+  }
+}
+
+private object WebSocketAccountServerEndpoint extends SparkzLogging {
+  var subscriptionCounter: AtomicInteger = new AtomicInteger(0)
+  var newHeadsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var newPendingTransactionsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var logsSubscriptions: util.ArrayList[SubscriptionWithFilter] = new util.ArrayList[SubscriptionWithFilter]()
+
+  val webSocketAccountChannelImpl = new WebSocketAccountChannelImpl()
+  private var walletAddresses: Set[Address] = webSocketAccountChannelImpl.getWalletAddresses
+  private var cachedBlocksReceipts: mutable.Queue[(String, Set[EthereumLogView])] = new mutable.Queue[(String, Set[EthereumLogView])]()
+  private val maxCachedBlockReceipts = 100
+
+  def notifySemanticallySuccessfulModifier(block: AccountBlock): Unit = {
+    log.info("Websocket received new block: "+block.toString)
+
+    val blockJson = webSocketAccountChannelImpl.accountBlockToWebsocketJson(block)
+
+    newHeadsSubscriptions.forEach(subscription => {
+      send(new WebSocketAccountEvent(params = new WebSocketAccountEventParams(subscription.subscriptionId, blockJson)), subscription.session)
+    })
+
+    //We have a chain reorganization
+    while (cachedBlocksReceipts.nonEmpty && !cachedBlocksReceipts.last._1.equals(block.parentId)) {
+      val oldTip = cachedBlocksReceipts.last
+      logsSubscriptions.forEach(subscription => {
+        val logsToSend = oldTip._2.filter(RpcFilter.testLog(subscription.filter.address, subscription.filter.topics))
+        sendTransactionLog(logsToSend.toSeq, subscription)
+      })
+      cachedBlocksReceipts = cachedBlocksReceipts.dropRight(cachedBlocksReceipts.size)
```

Comment:
Accordingly to this [documentation ](https://docs.scala-lang.org/overviews/collections-2.13/performance-characteristics.html) the Queue and the ListBuffer data structures have the same performance.
Btw I'm fine to replace the Queue with the ListBuffer if you prefer.

Reply:
let's change, just because we don't use it as a real Queue, especially here by dropping right.
PS: Better to use immutable List, than mutable ListBuffer (see the performance of `tail`)

Reply:
Done

Reply:
Done